treeherder/services/pulse/sources.py

```diff
@@ -0,0 +1,66 @@
+import environ
+
+env = environ.Env()
+
+
+def get_job_sources():
+    """
+    Get Job ingestion source locations.
+
+    Specifies the Pulse exchanges Treeherder will ingest data from for Job
+    data.  This list will be updated as new applications come online that
+    Treeherder supports. Treeherder will subscribe with routing keys that are
+    all combinations of ``project`` and ``destination`` in the form of:
+    <destination>.<project> Wildcards such as ``#`` and ``*`` are supported for
+    either field.
+    """
+    sources = env.json(
+        "PULSE_DATA_INGESTION_SOURCES",
+        default=[
+            {
+                "exchange": "exchange/taskcluster-treeherder/v1/jobs",
+                "projects": [
+                    '#'
+                    # some specific repos TC can ingest from
+                    # 'mozilla-central.#',
+                    # 'mozilla-inbound.#'
+                ],
+                "destinations": [
```

Comment:
In hindsight, the concept of ``destinations`` could probably be removed.  This was intended if there was some data that should go only to ``stage`` and not ``prod`` or vice versa.  But, in reality, I think we always want parity between the two.  This PR isn't the right place to make that change, but in the future we could explore whether to remove this.